bin/fetch.rs

```diff
@@ -74,11 +76,15 @@ fn main() -> Result<(), Box<dyn Error>> {
 
     // Populate bytecodes and state from RPC storage.
     let mut state = BTreeMap::<Address, EvmAccount>::new();
-    let mut bytecodes: BTreeMap<B256, EvmCode> = match File::open("data/bytecodes.bincode") {
-        Ok(file) => bincode::deserialize_from(BufReader::new(file))
-            .map_err(|err| format!("Failed to deserialize bytecodes from file: {err}"))?,
+    let mut bytecodes: BTreeMap<B256, EvmCode> = match File::open("data/bytecodes.bincode.gz") {
```

Comment:
Can you add a `// TODO: Deduplicate logic with [for_each_block_from_disk] when there is more usage`?